Fix #10853 - Fix colour editor for use in browsers that don't use -webkit- prefix

AJDBenner opened the pull request:
This patch will fix https://github.com/adobe/brackets/issues/10853. It adds all other browser prefixes in addition to the `-webkit-` prefix for the color editor. Here are some screenshots of what the changes look like.

Chrome
![chrome](https://cloud.githubusercontent.com/assets/9154982/7055375/e636c14c-de10-11e4-879a-3423cefc9f87.png)

Desktop Brackets 
![desktopbrackets](https://cloud.githubusercontent.com/assets/9154982/7055383/ec574524-de10-11e4-958a-9e20d7418bad.png)

Firefox
![firefox](https://cloud.githubusercontent.com/assets/9154982/7055387/f2c5d916-de10-11e4-87a0-29376f59b146.png)

Internet Explorer 10
![ie10](https://cloud.githubusercontent.com/assets/9154982/7055391/fa891cf8-de10-11e4-9b38-936a7e9de8f8.png)

Opera
![opera](https://cloud.githubusercontent.com/assets/9154982/7055395/ff905554-de10-11e4-9cb7-011f47bd5cc4.png)

Safari
![safari](https://cloud.githubusercontent.com/assets/9154982/7055396/0528bf2e-de11-11e4-8df2-17d1d35327bf.png)

AJDBenner committed:
Fix #150 - Fix colour editor for use in browsers that don't use -webkit- prefix

TomMalbran commented on src/extensions/default/InlineColorEditor/css/main.less:
```diff
@@ -136,9 +136,17 @@
+    background-image: -moz-linear-gradient(left, #FFFFFF 0%, rgba(255,255,255,0) 100%);
+    background-image: -o-linear-gradient(left, #FFFFFF 0%, rgba(255,255,255,0) 100%);
+    background-image: -ms-linear-gradient(left, #FFFFFF 0%, rgba(255,255,255,0) 100%);
+    background-image: linear-gradient(left, #FFFFFF 0%, rgba(255,255,255,0) 100%);
```
We don't really need to add any prefixed version of linear-gradient, so why not just leave the unprefixed version?

http://caniuse.com/#search=linear-gradient

valtlait replied:
Ln 142: It should be “to right”, not “left”

abose commented:
@le717 Bramble says brackets gradient picker could work in all browsers. 
NB; CEF2721 could land in Linux within 1.4.

AJDBenner commented:
I've tested using just the unprefixed version of linear-gradient and it works across all browsers. I also changed line 142 to be "to right" instead of "left". I'm unsure how to go about testing for CEF though. Any recommendations? Thanks for the comments by the way.

abose commented:
If it works in brackets , then shouldnt it work in CEF, as  brackets is essentially CEF.
Brackets as we assumes CEF could do away with the vendor prefixes, but if unprefixed gradient works, we could make the switch from -webkit-gradient to linear-gradient

AJDBenner commented:
I have changed this patch to include just the unprefixed version of linear-gradient. I also changed the direction from "left" to "to right" as suggested above by @valtlait. It is tested in Brackets and here is what it looks like.

![screen shot 2015-04-16 at 11 18 40 am](https://cloud.githubusercontent.com/assets/9154982/7184391/0418c99c-e42b-11e4-89d6-16b06cc9a213.png)

abose commented:
Thanks for updating quickly, release 1.3 would be next week, so tagging it for release 1.4

AJDBenner commented:
Sounds good. Thanks!

ficristo commented:
@AJDBenner could you update this PR?

valtlai commented:
`to bottom` can be removed since it is a default direction. Also `to top`, if the color stops are inverted.

valtlai commented on src/extensions/default/InlineColorEditor/ColorEditor.js:
```diff
@@ -171,8 +171,8 @@ define(function (require, exports, module) {
         this.$selectionBase.css("background-color", colorObject.toHexString());
-        this.$opacityGradient.css("background-image", "-webkit-gradient(linear, 0% 0%, 0% 100%, from(" + hueColor + "), to(transparent))");
-        
+        this.$opacityGradient.css("background-image", "linear-gradient(to bottom, " + hueColor + " 0%, rgba(0,0,0,0) 100%)");
```
`to bottom` and percent values can be removed:

```js
"linear-gradient(" + hueColor + ", rgba(0,0,0,0))"
```

valtlai commented on src/extensions/default/InlineColorEditor/css/main.less:
```diff
@@ -135,10 +135,10 @@
 }
 .color-editor section .color-selection-field .saturation-gradient {
-    background-image: -webkit-gradient(linear, 0% 50%, 100% 50%, from(#fff), to(rgba(255,255,255,0)));
+    background-image: linear-gradient(to right, #FFFFFF 0%, rgba(255,255,255,0) 100%);
```
`linear-gradient(to right, #fff, hsla(0,0%,100%,0))`

valtlai commented on src/extensions/default/InlineColorEditor/css/main.less:
```diff
@@ -135,10 +135,10 @@
 }
 .color-editor section .color-selection-field .luminosity-gradient {
-    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(transparent), to(#000));
+    background-image: linear-gradient(to top, #000000 0%, rgba(255,255,255,0) 100%);
```
`linear-gradient(hsla(0,0%,100%,0), #000)`

valtlai commented on src/extensions/default/InlineColorEditor/css/main.less:
```diff
@@ -200,7 +200,7 @@
 }
 .color-editor section .hue-slider {
-    background-image: -webkit-linear-gradient(top, #f00, #f0f, #00f, #0ff, #0f0, #ff0, #f00);
+    background-image: linear-gradient(to bottom, #f00, #f0f, #00f, #0ff, #0f0, #ff0, #f00);
```
Remove `to bottom`

valtlai commented:
I added some suggestions for simpler/shorter syntax. (See line notes.)

ficristo commented:
@valtlai since you have already looked at it do you want to set up a new PR for this?

valtlai commented:
@ficristo Here’s it: #12861

ficristo commented:
I close this in favour of #12861.
